draft-pda-protocol.md

```diff
@@ -1297,6 +1319,28 @@ participating aggregators stored unencrypted input shares on the same cloud
 object storage service, then that cloud vendor would be able to reassemble all
 the input shares and defeat privacy.
 
+#### Anti-replay {#anti-replay}
+
+Using a report in multiple batches, or multiple times within a single batch, is
+considered a privacy violation, since it may leak more information about that
+report than intended by the PDA protocol. (For example, this may violate
+differential privacy.) To mitigate this issue, the core specification imposes an
+ordering on reports so that aggregators can cheaply prevent replays attacks.
+
+Aggregate requests are ordered as follows: We say that a report `R2` *follows*
+report `R1` if either `R2.time > R1.time` or `R2.time == R1.time` and `R2.jitter
+> R1.jitter`. If `R2.time < R1.time`, or `R2.time == R1.time` but `R2.jitter <=
+R1.jitter`, then we say that `R2` *does not follow* `R1`.
+
+To prevent replay attacks, it suffices for each aggregator to ensure that each
+report it consumes follows the previous one it consumed. To prevent the
+adversary from tampering with the ordering of reports, honest clients
+incorporate the ordering-sensitive parameters `(time, jitter)` into the AAD for
+HPKE encryption.
+
+Note that this strategy may result in dropping reports that happen to have the
+same timestamp and jitter value.
```

Comment:
With 32 bit jitter, by the birthday paradox, we'd expect jitter collisions if a single task generates more than 2.1 billion reports in a second across all clients. That's definitely a lot, but why not make the jitter 64 bits so that it's astronomically unlikely?

Reply:
Actually, by the birthday paradox, the probability of a collision among just 2^16 reports is about 1.  For 64 bits, this same is true for 2^32 reports. If we want this probability to be negligible (i.e., about as unlikely as cracking AES-GCM, say) we would need 128 bits.

Personally, I'd be happy with 64 bits.

Reply:
Done.